client/X11/xf_event.c

```diff
@@ -982,7 +984,7 @@ static BOOL xf_event_PropertyNotify(xfContext* xfc, const XPropertyEvent* event,
 			if (status)
 			{
 				/* If the window is in the iconic state */
-				if (((UINT32)*prop == 3))
+				if (((UINT32)*prop == 3) && strcmp(getenv("DESKTOP_SESSION"), "gnome") != 0)
```

Comment:
sorry for the late comment, did miss this.
`getenv` might return `NULL` and `strcmp` does have undefined behavior with`NULL` arguments.

Reply:
maybe abstract this in `xf_utils.c/h` to generically query for some environment as you mentioned other gnome specific fixes to be done?

Reply:
@akallabeth Yes, sorry, I didn't consider that. I'll create an abstraction and fix the commit.